$home: mwm-title-bug.patch,v 1.1 2009/07/22 17:43:02 dukzcry Exp $

--- ResEncod.c.orig     Wed Jul 22 00:20:11 2009
+++ lib/Xm/ResEncod.c   Wed Jul 22 19:37:47 2009
@@ -776,10 +776,6 @@ XmCvtCTToXmString(
            } else {
                ok = False;
            }
-           if (ok) {
-             ctx->encoding = ctx->item;
-             ctx->encodinglen = ctx->itemlen;
-           }
            break;

        case CSI:
