$dukzcry$
--- orig/data.h	Wed Feb 28 17:21:58 2007
+++ my/data.h	Tue Jan  5 22:39:20 2010
@@ -13,6 +13,8 @@ bool XMP_ExtractTags_ID3v2(const char *data, char *art
 
 bool XMP_ExtractTags_WMA(const char *data, char *artist, char *title, char *album);
 bool XMP_ExtractTags_Other(const char *data, char *artist, char *title, char *album);
+/* Internet radio scrobbling */
+bool XMP_ExtractTags_NetRadio(const char *data, char *artist, char *title, char *album);
 bool XMP_ExtractTags_MBID(const char *data, char *mb);
 
 #endif
